### PR TITLE
td-uefi-pi: Fix panic issue in line 52 of src/hob.rs

### PR DESCRIPTION
**Pull request by liuw1**
Fix: #340
Signed-off-by: Wei Liu <wei3.liu@intel.com>

### CODE REVIEW

File: td-uefi-pi/src/hob.rs [hunk 1]
```diff
@@ -49,7 +49,10 @@ pub fn check_hob_length(hob: &[u8], hob_length: usize) -> Option<&[u8]> {
     if phit.header.r#type == HOB_TYPE_HANDOFF
         && phit.header.length as usize >= size_of::<HandoffInfoTable>()
     {
-        if hob_length as u64 == phit.efi_end_of_hob_list - hob.as_ptr() as u64 {
+        if u64::MAX - (hob_length as u64) < hob.as_ptr() as u64 {
```
Comment (jyao1) [inline td-uefi-pi/src/hob.rs:52]: Can we use rust checked_sub function to handle this instead of use u64::MAX?

Comment (liuw1) [reply td-uefi-pi/src/hob.rs]: Replaced

File: td-uefi-pi/src/hob.rs [hunk 2]
```diff
@@ -49,6 +49,7 @@ pub fn check_hob_length(hob: &[u8], hob_length: usize) -> Option<&[u8]> {
     if phit.header.r#type == HOB_TYPE_HANDOFF
         && phit.header.length as usize >= size_of::<HandoffInfoTable>()
     {
+        phit.efi_end_of_hob_list.checked_sub(hob.as_ptr() as u64)?;
```
Comment (jyao1) [inline td-uefi-pi/src/hob.rs:52]: I think we just need sub once. No need to sub twice.

Comment (liuw1) [reply td-uefi-pi/src/hob.rs]: Fixed

**Review by jyao1 — APPROVED** [review]

**Pull request merged**